src/main/java/eu/xenit/alfred/content/gateway/filter/web/ContentGridAppRequestWebFilter.java

```diff
@@ -0,0 +1,88 @@
+package eu.xenit.alfred.content.gateway.filter.web;
+
+import eu.xenit.alfred.content.gateway.routing.ServiceTracker;
+import eu.xenit.alfred.content.gateway.servicediscovery.ContentGridApplicationMetadata;
+import eu.xenit.alfred.content.gateway.servicediscovery.ContentGridDeploymentMetadata;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ServerWebExchange;
+import org.springframework.web.server.WebFilter;
+import org.springframework.web.server.WebFilterChain;
+import reactor.core.publisher.Mono;
+
+@Slf4j
+@RequiredArgsConstructor
+public class ContentGridAppRequestWebFilter implements WebFilter {
+
+    /**
+     * This {@link WebFilter} needs to be registered before the Spring Security WebFilter, so that authentication
+     * and authorization infrastructure can use request attributes set by this WebFilter. The Spring Security
+     * WebFilter is registered by {@code WebFluxSecurityConfiguration}.
+     *
+     * @see <a href="https://github.com/spring-projects/spring-boot/issues/33551">Spring Boot issue 33551</a>
+     *
+     */
+    public static final int CONTENTGRID_WEB_FILTER_CHAIN_FILTER_ORDER = -200;
+
+    public static final String CONTENTGRID_SERVICE_INSTANCE_ATTR = "com.contentgrid.gateway/service-instance";
```

Comment:
to give it some namespace, slightly analogues with the kubernetes labels
I've seen full classnames used as namespace, but that feels awkward to me personally

Reply:
You want to avoid matching the k8s labels exactly to avoid confusion?

To be honest, I would then just call it `com.contentgrid.gateway.service-instance`